BUILD

```diff
@@ -534,6 +534,25 @@ GRPCXX_PUBLIC_HDRS = [
     "include/grpcpp/impl/codegen/sync.h",
 ]
 
+grpc_cc_library(
```

Comment:
Why need a new library?

Reply:
So that code can be placed outside of src/core and let `grpc` and `grpc_unsecure` depends on it using its build system.
Anyway, what's your suggestion?

Reply:
Why not put the source in the source list just like CMakeLists.txt does?

Reply:
CMakeLists.txt is generated from BUILD, upstream may overwrite CMakeLists.txt if they add or remove a file.

Reply:
@BusyJay PTAL, thanks!

Reply:
I'm a bit confused. Now is CMakeLists.txt auto-generated? Note we always need to modify CMakeLists.txt as the generated one from upstream doesn't always compile.